src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/ReadOrientationArtifactUnitTest.java

```diff
@@ -0,0 +1,211 @@
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+public class ReadOrientationArtifactUnitTest extends GATKBaseTest {
+    final int chromosomeIndex = 1;
+    final int variantPosition = 100_000;
+    final String sampleName = "someTumor";
+    final int readLength = 9;
+
+    @DataProvider(name = "readOrientationArtifact")
+    public Object[][] readOrientationArtifactData() {
+        return new Object[][]{
+                {150, 0.5, 0.5, 0.0, 1e-3, Optional.empty()}, // a likely germline het with no bias
+                {150, 0.3, 0.5, 0.0, 1e-3, Optional.empty()}, // allele fraction 0.3 variant with no bias
+                {150, 0.3, 0.99, 1.0, 1e-3, Optional.of(ReadOrientation.F1R2)}, // allele fraction 0.3 variant with a heavy F1R2 bias
+                {150, 0.3, 0.06, 1.0, 1e-3, Optional.of(ReadOrientation.F2R1)}, // allele fraction 0.2 variant with a moderate F2R1 bias
+                {150, 0.2, 0.01, 1.0, 1e-2, Optional.of(ReadOrientation.F2R1)}, // allele fraction 0.2 variant with a heavy F2R1 bias
+                {150, 0.03, 0.0, 1.0, 1e-1, Optional.of(ReadOrientation.F2R1)}, // low allele fraction with a heavy F2R1 bias
+                {300, 0.02, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)}, // low allele fraction, high depth
+                {200, 0.02, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)}, // low allele fraction, high depth - NOTE: to get this case right, *maybe* we should learn f.
+                {700, 0.025, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)}, // high depth hom ref site with some error
+                {50, 0.01, 0.8, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)}, // medium depth hom ref site with some error
+                {160, 0.025, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)},
+                {150, 0.025, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)},
+                {700, 0.01, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)}};
+    }
+
+    @DataProvider(name = "failingCases")
+    public Object[][] failingCasesData() {
+        return new Object[][]{
+                {72, 0.041, 1.0, 1.0, 1e-1, Optional.of(ReadOrientation.F1R2)}};
+    }
+
+    @Test(dataProvider="readOrientationArtifact")
+    public void test(final int depth, final double alleleFraction, final double altF1R2Fraction,
+                     final double expectedArtifactProb, final double epsilon,
+                     final Optional<ReadOrientation> expectedArtifactType) throws IOException {
+        // Why is it that the annotations get larger window than the walker (CollectData?)
```

Comment:
What's this comment about?

Reply:
I think it can safely be ignored. Removed